llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -51,26 +51,32 @@ void dumpInterBlock(const InterBlockEdges &Edges) {
   }
 }
 
-void emitBundlesTopDown(const std::vector<MachineBundle> &Bundles,
-                        ResourceScoreboard<FuncUnitWrapper> &Scoreboard,
-                        AIEHazardRecognizer *HR) {
+void emitBundlesTopDown(
+    const std::vector<MachineBundle> &Bundles,
+    std::map<MachineInstr *, const MCInstrDesc *> SelectedAltDesc,
+    ResourceScoreboard<FuncUnitWrapper> &Scoreboard, AIEHazardRecognizer *HR) {
 
   const int TotalBundles = Bundles.size();
   const int AmountToEmit = std::min(TotalBundles, HR->getConflictHorizon());
   // Do not emit more than the specified by the conflict horizon. More
   // then this will not cause conflicts.
   for (int i = TotalBundles - AmountToEmit; i < TotalBundles; i++) {
-    for (MachineInstr *MI : Bundles[i].getInstrs())
-      HR->emitInScoreboard(Scoreboard, MI->getDesc(), HR->getMemoryBanks(MI),
+    for (MachineInstr *MI : Bundles[i].getInstrs()) {
+      auto Desc = SelectedAltDesc.find(MI) != SelectedAltDesc.end()
```

Comment:
I think this whole AltDesc handling should be in a different class/file. This file is a bit bloated and we share this manipulation logic across here and HazardRecognizer.

Reply:
Thanks for the guidance have accordingly created #193